File: backend/app/schemas/memory.py
# backend/app/schemas/memory.py — Pydantic схемы памяти
"""
Схемы для памяти пользователя.
"""

from datetime import datetime

from pydantic import BaseModel, Field


class MemoryIn(BaseModel):
    """Добавление записи памяти."""
    type: str = Field(pattern="^(profile|preference|shared_story)$")
    text: str = Field(min_length=1, max_length=2000)
    importance: int = Field(ge=1, le=5, default=3)


class MemoryOut(BaseModel):
    """Запись памяти."""
    id: int
    type: str
    text: str
    importance: int
    created_at: datetime
    updated_at: datetime | None

File: frontend/src/api/memory.ts
// frontend/src/api/memory.ts — методы памяти
// Назначение: list/add/delete

import { apiFetch } from "./http";

export type MemoryOut = {
  id: number;
  type: "profile" | "preference" | "shared_story";
  text: string;
  importance: number;
  created_at: string;
  updated_at: string | null;
};

export function memoryList(): Promise<MemoryOut[]> {
  return apiFetch<MemoryOut[]>("/api/v1/memory");
}

export function memoryAdd(type: MemoryOut["type"], text: string, importance = 3): Promise<MemoryOut> {
  return apiFetch<MemoryOut>("/api/v1/memory", {
    method: "POST",
    body: JSON.stringify({ type, text, importance }),
  });
}

export function memoryDelete(id: number): Promise<void> {
  return apiFetch<void>(`/api/v1/memory/${id}`, { method: "DELETE" });
}
